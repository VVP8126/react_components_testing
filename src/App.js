import './App.css';
import { Link, Routes, Route } from 'react-router-dom';
import About from './pages/About';
import Main from './pages/Main';
import Page404 from './pages/Page404';
import UserDetailedPage from './pages/UserDetailedPage';
import Counter from './components/counter/Counter';

function App() {
  return (
    <div>
      <div className='menu_block'>
        <Link data-testid="main-page-link"  className='menu_item' to={"/"} >MAIN</Link>
        <Link data-testid="redux-page-link" className='menu_item' to={"/counter"} >REDUX</Link>
        <Link data-testid="about-page-link" className='menu_item' to={"/about"} >ABOUT</Link>
      </div>
      <Routes>
        <Route path='/about' element={<About />} />
        <Route path='/about/:id' element={<UserDetailedPage />} />
        <Route path='/counter' element={<Counter />} />
        <Route path='/' element={<Main />} />
        <Route path='/*' element={<Page404 />} />
      </Routes>
    </div>
  );
}

export default App;
