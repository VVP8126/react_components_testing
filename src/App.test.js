import { render, screen } from '@testing-library/react';
import App from "./App";
import userEvent from '@testing-library/user-event';
import { MemoryRouter } from "react-router-dom";

describe("Start page Test", () => {
  
  beforeEach(() => {
    render(<MemoryRouter initialEntries={["/fake-path"]} ><App /></MemoryRouter>);
  });

  test('Test existing routings', () => {
    const linkMainElement = screen.getByTestId("main-page-link");
    expect(linkMainElement).toBeInTheDocument();
    const linkAboutElement = screen.getByTestId("about-page-link");
    expect(linkAboutElement).toBeInTheDocument();
    userEvent.click(linkMainElement); // Click at first (left) link
    expect(screen.getByTestId("main-page")).toBeInTheDocument();
    userEvent.click(linkAboutElement); // Click at second (right) link - passing to <About /> element
    expect(screen.getByTestId("about-page")).toBeInTheDocument();
    userEvent.click(linkMainElement); // Click at first (left) link - return to <Main /> element
    expect(screen.getByTestId("main-page")).toBeInTheDocument();
  });

  test('Test not-existing routing', () => {
    expect(screen.getByTestId("page-not-found")).toBeInTheDocument();
  });

});
