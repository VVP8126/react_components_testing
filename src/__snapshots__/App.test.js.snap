// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start page Test Test link render 1`] = `
<a
  class="App-link"
  href="https://reactjs.org"
  rel="noopener noreferrer"
  target="_blank"
>
  Learn React
</a>
`;

exports[`renders learn react link 1`] = `
<a
  class="App-link"
  href="https://reactjs.org"
  rel="noopener noreferrer"
  target="_blank"
>
  Learn React
</a>
`;
